spec/robots/dor_repo/ocr/fetch_files_spec.rb

```diff
@@ -0,0 +1,154 @@
+  let(:structural) do
+    {
+      contains: [
+        {
+          type: 'https://cocina.sul.stanford.edu/models/resources/image',
+          externalIdentifier: 'bb111bb2222_1',
+          label: 'Image 1',
+          version: 1,
+          structural: {
+            contains: [
+              {
+                type: 'https://cocina.sul.stanford.edu/models/file',
+                externalIdentifier: 'https://cocina.sul.stanford.edu/file/d30f48f9-1c11-4290-95a4-fea64e346db9',
+                label: 'image111.tif',
+                filename: 'image111.tif',
+                size: 123,
+                version: 1,
+                hasMimeType: 'image/tiff',
+                hasMessageDigests: [
+                  {
+                    type: 'md5',
+                    digest: '42616f9e6c1b7e7b7a71b4fa0c5ef794'
+                  }
+                ],
+                access: {
+                  view: 'dark',
+                  download: 'none',
+                  controlledDigitalLending: false
+                },
+                administrative: {
+                  publish: false,
+                  sdrPreserve: true,
+                  shelve: false
+                }
+              }
+            ]
+          }
+        },
+        {
+          type: 'https://cocina.sul.stanford.edu/models/resources/image',
+          externalIdentifier: 'bb111bb2222_2',
+          label: 'Image 2',
+          version: 1,
+          structural: {
+            contains: [
+              {
+                type: 'https://cocina.sul.stanford.edu/models/file',
+                externalIdentifier: 'https://cocina.sul.stanford.edu/file/39a83c02-5d05-4c3d-bff1-080772cfdd99',
+                label: 'image112.tif',
+                filename: 'image112.tif',
+                size: 123,
+                version: 1,
+                hasMimeType: 'image/tiff',
+                hasMessageDigests: [
+                  {
+                    type: 'sha1',
+                    digest: '5c9f6dc2ca4fd3329619b54a2c6f99a08c088444'
+                  },
+                  {
+                    type: 'md5',
+                    digest: 'ac440802bd590ce0899dafecc5a5ab1b'
+                  }
+                ],
+                access: {
+                  view: 'dark',
+                  download: 'none',
+                  controlledDigitalLending: false
+                },
+                administrative: {
+                  publish: false,
+                  sdrPreserve: true,
+                  shelve: false
+                }
+              }
+            ]
+          }
+        }
+      ],
+      hasMemberOrders: [],
+      isMemberOf: []
+    }
+  end
+
+  let(:dsa_object_client) do
+    instance_double(Dor::Services::Client::Object, find: cocina_model, update: true)
+  end
+
+  let(:pres_client) do
+    instance_double(Preservation::Client, objects: objects_client)
+  end
+
+  let(:objects_client) do
+    instance_double(Preservation::Client::Objects)
+  end
+
```

Comment:
I needed to mock out workflow_client and other things following in order for the `workflow.context` step to work. If there's an easier way around this let me know. Also, we may need similar code in other workflow tests, which might make copy & pasting this a bit tedious?